### PR TITLE
Validate that control-only Instance nodes cannot change IG membership

### PR DESCRIPTION
**Pull request by jbradberry**
##### SUMMARY
Backend part of #10853 

related #10830 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 19.3.0
```

### CODE REVIEW

File: awx/api/views/mixin.py [hunk 1]
```diff
@@ -68,13 +68,23 @@ class InstanceGroupMembershipMixin(object):
...
+        return sub_id, res
+
     def attach(self, request, *args, **kwargs):
         response = super(InstanceGroupMembershipMixin, self).attach(request, *args, **kwargs)
         sub_id, res = self.attach_validate(request)
         if status.is_success(response.status_code):
             if self.parent_model is Instance:
-                ig_obj = get_object_or_400(self.model, pk=sub_id)
-                inst_name = ig_obj.hostname
+                inst_name = self.get_parent_object().hostname
```
Comment (jbradberry) [inline awx/api/views/mixin.py:87]: I should note that this block here appears to have been a bug.  Instance groups do not have a hostname.

**Review by AlanCoding — APPROVED** [review]
nice relevant unit tests

**Comment by jbradberry** [comment]
YOLO http://jenkins.ansible.eng.rdu2.redhat.com/blue/organizations/jenkins/Test_Tower_Yolo_Express/detail/Test_Tower_Yolo_Express/25488/pipeline

**Comment by jbradberry** [comment]
http://jenkins.ansible.eng.rdu2.redhat.com/blue/organizations/jenkins/Test_Tower_Yolo_Express/detail/Test_Tower_Yolo_Express/25669/pipeline

**Comment by jbradberry** [comment]
Current yolo run: (removed)

**Review by CFSNM — APPROVED** [review]

**Review by Zokormazo — APPROVED** [review]

**Pull request merged**